x-pack/legacy/plugins/siem/public/components/ml/api/error_to_toaster.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { isError } from 'lodash/fp';
+import uuid from 'uuid';
+import { ActionToaster, AppToast } from '../../toasters';
+import { ToasterErrorsType, ToasterErrors } from './throw_if_not_ok';
+
+export type ErrorToToasterArgs = Partial<AppToast> & {
+  error: unknown;
+  dispatchToaster: React.Dispatch<ActionToaster>;
+};
+
+export const errorToToaster = ({
+  id = uuid.v4(),
+  title,
+  error,
+  color = 'danger',
+  iconType = 'alert',
+  dispatchToaster,
+}: ErrorToToasterArgs) => {
+  if (isToasterError(error)) {
```

Comment:
nit,  I think that you can simplify it like that
```
const toast: AppToast = {
  id,
  title,
  color,
  iconType,
  errors: ['Network Error'],
};

if (isToasterError(error)) {
  toast.errors = error.messages;
} else if (isAnError(error)) {
  toast.errors = [error.message];
}

dispatchToaster({
  type: 'addToaster',
  toast,
});
```

Reply:
Appreciate it, I will try to work it into a follow up PR 🤞 here soon.

Thank you for looking at my PR Xavier and giving me feedback.